compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -2367,25 +2367,25 @@ private void createVarRefAssignmentStmts(BLangTupleVarRef parentTupleVariable, B
      * This method creates a assignment statement and assigns and array expression based on the given indexExpr.
      *
      */
-    private void createSimpleVarRefAssignmentStmt(BLangVariableReference simpleVarRef, BLangBlockStmt parentBlockStmt,
-                                                  BLangExpression indexExpr, BVarSymbol tupleVarSymbol,
-                                                  BLangIndexBasedAccess parentArrayAccessExpr) {
+    private void createAssignmentStmt(BLangAccessExpression accessExpression, BLangBlockStmt parentBlockStmt,
```

Comment:
Shouldn't this be a `BLangAccessibleExpression`?

Reply:
Fixed in 004d3a3